src/pyflex/window_selector.py

```diff
@@ -316,25 +339,28 @@ def select_windows(self):
 
         self.calculate_preliminiaries()
 
+        self.determine_signal_and_noise_indices()
+        if self.config.check_global_data_quality:
+            if not self.check_data_quality():
+                return []
+
```

Comment:
Good idea :-) 

You might have to a bit careful as signal and noise indices cannot be determined if either event or station information is missing. But the tests should catch that.